### PR TITLE
WIP: Set time_viewer and show_traces to auto

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR sets `time_viewer='auto'` and `show_traces='auto'`. It's still a work in progress.

It's an item of #7162

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
To achieve `time_viewer=auto`, the following 3d viz tests need to be ported to the `renderer_interactive` fixture:
* test_plot_sparse_source_estimates
* test_process_clim_plot

**Comment by GuillaumeFavelier** [comment]
By default the 3d tests are done with `off_screen=True` to avoid opening a window but in order to use the `PyVista` widgets, `off_screen` should be disabled (to obtain an interactive window).
The goal of the `renderer_interactive` fixture is just that.

**Comment by GuillaumeFavelier** [comment]
The latest commit is marked as test because the `renderer_interactive` fixture does not include `mayavi` for now. It needs to be fixed.

**Comment by GuillaumeFavelier** [comment]
The `interactive` parameter of `_testing_context()` is unused in the `_pysurfer_mayavi` 3d backend:

https://github.com/mne-tools/mne-python/blob/499f917dd301826606f706e775c611c3e215d1b0/mne/viz/backends/_pysurfer_mayavi.py#L464-L472

compared to `_pyvista`:

https://github.com/mne-tools/mne-python/blob/499f917dd301826606f706e775c611c3e215d1b0/mne/viz/backends/_pyvista.py#L647-L661

So I think it's totally safe to add it to the list:

https://github.com/mne-tools/mne-python/blob/499f917dd301826606f706e775c611c3e215d1b0/mne/conftest.py#L238-L243

**Comment by GuillaumeFavelier** [comment]
The tests using `renderer_interactive` are:

```
test_brain_timeviewer
test_brain_timeviewer_traces
test_brain_linkviewer
test_plot_sparse_source_estimates
test_process_clim_plot
test_link_brains
```

Among those, only:

```
test_plot_sparse_source_estimates
test_process_clim_plot
```

are supported by `mayavi`, the rest should raise `NotImplementedError`.

**Comment by GuillaumeFavelier** [comment]
I forgot to update Circle config yesterday... The issue with pytest-sugar is back

**Comment by GuillaumeFavelier** [comment]
Oh no. It's actually another release of pytest:

image

**Comment by GuillaumeFavelier** [comment]
Can we just do `pytest<5.4.0` for now?

**Comment by GuillaumeFavelier** [comment]
I notice a few warnings of the type:

```
UserWarning: Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.
```

Also, there is an issue with `time_label`:

```
AttributeError: 'TimeSlider' object has no attribute 'time_label'
```

I'll push a fix.

**Comment by GuillaumeFavelier** [comment]
Waiting for https://github.com/mne-tools/mne-python/pull/7257 before merging conflicts.

**Comment by GuillaumeFavelier** [comment]
The current issue is a timeout of Circle during doc building a bit after the visualization folder.

**Comment by larsoner** [comment]
Merge conflict

**Comment by GuillaumeFavelier** [comment]
Since #7458 is merged, the workaround scraper needs to be removed. I'll fix this

File: mne/viz/tests/test_3d.py [hunk 1]
```diff
@@ -137,11 +137,8 @@ def test_plot_sparse_source_estimates(renderer):
     stc_data[1, 4] = 2.
     vertices = [vertices[inds], np.empty(0, dtype=np.int)]
     stc = SourceEstimate(stc_data, vertices, 1, 1)
-    surf = plot_sparse_source_estimates(sample_src, stc, bgcolor=(1, 1, 1),
-                                        opacity=0.5, high_resolution=False)
-    if renderer.get_3d_backend() == 'mayavi':
-        import mayavi  # noqa: F401 analysis:ignore
-        assert isinstance(surf, mayavi.modules.surface.Surface)
```
Comment (larsoner) [inline mne/viz/tests/test_3d.py]: Shouldn't we keep some variant of this? Can you instead change it to `renderer_interactive.get_3d_backend()`?

Comment (GuillaumeFavelier) [reply mne/viz/tests/test_3d.py]: Indeed that was a bit too agressive, I'll add a branch for mayavi

**Comment by GuillaumeFavelier** [comment]
I'll try `[circle full]` once https://github.com/mne-tools/mne-python/pull/7480 is merged

**Comment by larsoner** [comment]
Merged @GuillaumeFavelier

**Comment by GuillaumeFavelier** [comment]
#7480 was not enough apparently since we got a timeout too, I'm trying with `show_traces=False` and at the same time see if I can improve the memory footprint of `_Brain`

**Comment by GuillaumeFavelier** [comment]
So far my changes seem to help freeing memory but nothing big for `_Brain` (`master` in black, `local changes` in blue)

image

**Comment by GuillaumeFavelier** [comment]
@agramfort the timeout seems related to `show_traces=True` because now CircleCI passes:

https://18788-1301584-gh.circle-artifacts.com/0/dev/index.html

Is it good enough to have only `time_viewer='auto'`?

**Comment by agramfort** [comment]
yes ok. It can be improved later after the release.

>

**Comment by GuillaumeFavelier** [comment]
This is ready for reviews then

**Comment by larsoner** [comment]
> > Is it good enough to have only time_viewer='auto'?

> yes ok. It can be improved later after the release.

I'm not so sure because this would be a change in behavior with no deprecation cycle.

A lesser evil would probably be to add a private `_MNE_TRACES_AUTO=false` env var that we set in CircleCI and detect in `_brain` to make it so `'auto'` means False. This would make the output different from what a user would see when they ran the script with the PyVista backend and more like what it would be like with the Mayavi backend -- but I doubt anyone would really notice.

**Comment by GuillaumeFavelier** [comment]
> A lesser evil would probably be to add a private _MNE_TRACES_AUTO=false env var

I guess it's possible indeed, what do you think @agramfort ?

Why private though @larsoner ?

**Comment by larsoner** [comment]
> Why private though @larsoner ?

Hopefully eventually we can fix the memory/timeout problem and kill it entirely. We don't want it actually to be a permanent solution (or give users yet another way to set things, they should use `kwargs`)

**Comment by GuillaumeFavelier** [comment]
> Hopefully eventually we can fix the memory/timeout problem and kill it entirely

I'll continue to investigate locally

**Comment by larsoner** [comment]
Yes please do, but in the meantime the private env var workaround would make it so that we don't have to figure it out before release

**Comment by larsoner** [comment]
Pushed a commit with the CircleCI workaround and setting both `show_traces` and `time_viewer` to `'auto'` as default, let's see if it gets us around the CI problem

**Comment by GuillaumeFavelier** [comment]
Thank you @larsoner

**Comment by agramfort** [comment]
if @drammock and @hoechenberger are happy I am happy. Happy @drammock and @hoechenberger ?

**Comment by hoechenberger** [comment]
LGTM!

**Pull request merged**